### DISCUSSION TITLE
[data imported in Docker] 401 if using Confidential checkbox unselected

### DISCUSSION THREAD

**Discussion by kyxap** [firefly-iii/firefly-iii#7744] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Yeah it could be me but let me try to explain. 

I was able to Authorize data-importer when im **not** using **Confidential** checkbox but I can’t auth if that checkbox selected.

<img width="512" alt="image" src="https://github.com/firefly-iii/firefly-iii/assets/3080529/6e3d3be7-4e31-45c4-8002-87fad3ca52e3">

After I've put url and client id im getting: (see debug)

for me its not clear how persona token supposed to be wired with data-importer, i've added that token to the imported env but does not seems resolving the issue or maybe im missing something. 

### Debug information

data importer ui:
```html
Error :(
The server you tried to connect to responsed with an error.

Client error: `POST http://192.168.100.200:9191/oauth/token` resulted in a `401 Unauthorized` response: {"error":"invalid_client","error_description":"Client authentication failed","message":"Client authentication failed"}
```

docker logs:
```bash
firefly_iii_importer  | [2023-07-10 04:38:47] local.ERROR: Response from server: "{"error":"invalid_client","error_description":"Client authentication failed","message":"Client authentication failed"}"
firefly_iii_importer  | 172.29.0.1 - - [10/Jul/2023:04:38:47 +0200] "GET /callback?code=<CODE_REMOVED> HTTP/1.1" 200 1988 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
```

### Expected behaviour

OK and Redirect to page with imports

### Steps to reproduce

- install firefly-iii with data importer
- create new client id (select Confidential)
- I also created new personal token and added it to the imported .env config (FIREFLY_III_ACCESS_TOKEN=eyJ0eXAi......) based on documentation
- removed all containers and did the `>docker compose` coz I didn't get it how to add token after containers are build
- login to the app again
- go to data importer url and enter firefly url and just created id 
- click authorize
See error

**Comment by JC5** [comment]
I tried this yesterday for something unrelated and it worked fine. 

Did you press [Reauthenticate] in between tries?

**Comment by kyxap** [comment]
yeah I tried to press Reauthenticate couple times - same issue.

**Comment by JC5** [comment]
> I was able to Authorize data-importer when im not using Confidential checkbox but I can’t auth if that checkbox selected.

I'm reading this again. This is expected behaviour. NOT selecting the checkbox is how it's supposed to work.

https://docs.firefly-iii.org/data-importer/installation/configuration/#access-to-the-api
